clients/src/test/java/org/apache/kafka/clients/consumer/internals/ShareMembershipManagerTest.java

```diff
@@ -1137,10 +1137,25 @@ public void testRevokePartitionsUsesTopicNamesLocalCacheWhenMetadataNotAvailable
     }
 
     @Test
-    public void testOnSubscriptionUpdatedTransitionsToJoiningOnlyIfNotInGroup() {
+    public void testOnSubscriptionUpdatedDoesNotSetShouldTransitionToJoiningIfInGroup() {
```

Comment:
let's sure make sure to align these with the names in the consumer tests if they change with the comments above

Reply:
Yes, align test case names between `ConsumerMembershipManagerTest` and `ShareMembershipManagerTest`.